crypto/bn256/cloudflare/gfp.go

```diff
@@ -79,3 +80,130 @@ func (e *gfP) Unmarshal(in []byte) error {
 
 func montEncode(c, a *gfP) { gfpMul(c, a, r2) }
 func montDecode(c, a *gfP) { gfpMul(c, a, &gfP{1}) }
+
+func isZero(a *gfP) bool {
+	return a[0] == 0 && a[1] == 0 && a[2] == 0 && a[3] == 0
```

Comment:
```suggestion
	return (a[0] | a[1] | a[2] | a[3]) == 0
```

Reply:
This code is not in a hot path, so I can either keep it more readable (current form) or exploit bit manipulations, it's up to you.

Reply:
Personally I prefer the shorter form regardless